functions.sh.in

```diff
@@ -52,6 +52,39 @@ hostspec_to_libc() {
 	esac
 }
 
+_print_repository_arg() {
+	local format='--repository %s'
+	local path="$1"
+
+	if [ $# -eq 1 ]; then
+		printf -- "$format" "$@"
+	else
+		shift
+		printf -- "$format " "$path/$1"
+		shift
+		if [ $# -gt 1 ]; then
+			_print_repository_arg "$path" "$@"
+		else
+			_print_repository_arg "$@"
+		fi
+	fi
+}
+
+repo_to_repositories() {
+	local path="$1"
+	
+	while [ "$path" != "${path%/}" ]; do
+		path="${path%/}"
+	done
+	
+	case "$path" in
+	*/main)			_print_repository_arg "$path" ;;
+	*/community)	_print_repository_arg "${path%/*}" main "$path" ;;
+	*/testing)		_print_repository_arg "${path%/*}" main community "$path" ;;
+	*)				_print_repository_arg "${path%/*}" main community testing "$path" ;;
+	esac
```

Comment:
we should use the `.rootbld-repositories` file instead of hardcode the repository dependencies in the abuild code.

Or at least share common code with the `rootbld` logic for this.

Reply:
I don't see a good way to use that file. We could get rid of it with a common function instead.

Reply:
The key point here is that the dependencies between repositories are stored in a "config" file in aports tree itself. So if we rename `main` to `core` or add another repository or similar, then we dont depend on fixing the abuild code to make it work.

@kunkku do you have any suggestion how to do this properly?

Reply:
@ncopa @kunkku I added some logic to try to parse `.rootbld-repositories` files when they can be found.